### ISSUE TITLE
🐛 [Crash] CLuaTradeContainer [commit 1c56ff0]

### ISSUE THREAD

**Issue by kaincenteno** [DerpyProjectGroup/topaz#2040]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)

## Branch affected by issue 

`base`

```
[06/18/22 04:26:34:151][map][info][info] parse: 036 | 0B47 0B45 20 from user: Midknight (parse:720)
terminate called without an active exception

Thread 1 "xi_map" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007fbdffbe6859 in __GI_abort () at abort.c:79
#2  0x00007fbdfffe3911 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007fbdfffef38c in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007fbdfffef3f7 in std::terminate() () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000000000040e2ab in __clang_call_terminate ()
#6  0x00000000005b6090 in sol::detail::static_trampoline<&sol::function_detail::upvalue_this_member_function<CLuaBaseEntity, bool (CLuaBaseEntity::*)(sol::variadic_args)>::real_call> (L=0x41b82378) at /server/ext/sol/include/sol/sol.hpp:8757
#7  sol::detail::typed_static_trampoline<int (*)(lua_State*), &sol::function_detail::upvalue_this_member_function<CLuaBaseEntity, bool (CLuaBaseEntity::*)(sol::variadic_args)>::real_call> (L=0x41b82378) at /server/ext/sol/include/sol/sol.hpp:8879
#8  sol::function_detail::upvalue_this_member_function<CLuaBaseEntity, bool (CLuaBaseEntity::*)(sol::variadic_args)>::call<false, false> (L=0x41b82378) at /server/ext/sol/include/sol/sol.hpp:19332
#9  0x00007fbe00186fb7 in ?? () from /lib/x86_64-linux-gnu/libluajit-5.1.so.2
#10 0x00007fbe001d54a4 in lua_pcall () from /lib/x86_64-linux-gnu/libluajit-5.1.so.2
#11 0x0000000000683ff3 in sol::basic_protected_function<sol::stack_reference, true, sol::basic_reference<false> >::luacall<true> (this=0x7ffe4d523ca0, argcount=4, result_count_=-1, h=...) at /server/ext/sol/include/sol/sol.hpp:21042
#12 sol::basic_protected_function<sol::stack_reference, true, sol::basic_reference<false> >::invoke<true>(sol::types<>, std::integer_sequence<unsigned long>, long, sol::detail::protected_handler<true, sol::basic_reference<false> >&) const (this=0x7ffe4d523ca0, n=4, h=...) at /server/ext/sol/include/sol/sol.hpp:21073
#13 sol::basic_protected_function<sol::stack_reference, true, sol::basic_reference<false> >::call<, CLuaBaseEntity, CLuaBaseEntity, CLuaTradeContainer, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [8]> >&>(CLuaBaseEntity&&, CLuaBaseEntity&&, CLuaTradeContainer&&, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [8]> >&) const (this=0x7ffe4d523ca0, args=..., args=..., args=..., args=...) at /server/ext/sol/include/sol/sol.hpp:20981
#14 sol::table_proxy<sol::basic_table_core<true, sol::basic_reference<false> >&, std::tuple<char const (&) [3], char const (&) [8], char const (&) [12], char const (&) [19], char const (&) [8]> >::call<, CLuaBaseEntity, CLuaBaseEntity, CLuaTradeContainer, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [8]> >&>(CLuaBaseEntity&&, CLuaBaseEntity&&, CLuaTradeContainer&&, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [8]> >&) (this=<optimized out>, args=..., args=..., args=..., args=...) at /server/ext/sol/include/sol/sol.hpp:24839
#15 sol::table_proxy<sol::basic_table_core<true, sol::basic_reference<false> >&, std::tuple<char const (&) [3], char const (&) [8], char const (&) [12], char const (&) [19], char const (&) [8]> >::operator()<CLuaBaseEntity, CLuaBaseEntity, CLuaTradeContainer, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [8]> >&> (this=<optimized out>, args=..., args=..., args=..., args=...) at /server/ext/sol/include/sol/sol.hpp:24844
#16 luautils::OnTrade (PChar=<optimized out>, PNpc=<optimized out>) at /server/src/map/lua/luautils.cpp:2086
#17 0x00000000007d106f in SmallPacket0x036 (PSession=<optimized out>, PChar=0x7fbdf67f45d0, data=...) at /server/src/map/packet_system.cpp:1749
#18 0x0000000000802f20 in std::_Function_handler<void (map_session_data_t*, CCharEntity*, CBasicPacket), void (*)(map_session_data_t*, CCharEntity*, CBasicPacket)>::_M_invoke(std::_Any_data const&, map_session_data_t*&&, CCharEntity*&&, CBasicPacket&&) (__functor=..., __args=..., __args=..., __args=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/std_function.h:300
#19 0x00000000007a9f4a in std::function<void (map_session_data_t*, CCharEntity*, CBasicPacket)>::operator()(map_session_data_t*, CCharEntity*, CBasicPacket) const (this=<optimized out>, __args=..., __args=..., __args=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/std_function.h:688
#20 parse (buff=0x7fbdf6172360 "G\vE\v\374\002", buffsize=<optimized out>, from=<optimized out>, map_session_data=0x7fbdf661b790) at /server/src/map/map.cpp:746
#21 do_sockets (rfd=<optimized out>, next=...) at /server/src/map/map.cpp:512
#22 0x000000000041deab in main (argc=1, argv=<optimized out>) at /server/src/common/kernel.cpp:278
```

**Comment by kaincenteno** [comment]
Crash after genkai 1

1. Finish Limit Break 1
2. Talk to Maat again without zoning/leveling up
3. crash

```
Thread 1 "xi_map" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007f4561d28859 in __GI_abort () at abort.c:79
#2  0x00007f4562125911 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007f456213138c in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007f45621313f7 in std::terminate() () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000000000040e2fb in __clang_call_terminate ()
#6  0x00000000005a4d7f in sol::detail::static_trampoline<&sol::function_detail::upvalue_this_member_function<CLuaBaseEntity, void (CLuaBaseEntity::*)(CLuaBaseEntity*, unsigned short, sol::basic_object<sol::basic_reference<false> > const&, sol::basic_object<sol::basic_reference<false> > const&)>::real_call> (L=0x4123e378) at /server/ext/sol/include/sol/sol.hpp:8757
#7  sol::detail::typed_static_trampoline<int (*)(lua_State*), &sol::function_detail::upvalue_this_member_function<CLuaBaseEntity, void (CLuaBaseEntity::*)(CLuaBaseEntity*, unsigned short, sol::basic_object<sol::basic_reference<false> > const&, sol::basic_object<sol::basic_reference<false> > const&)>::real_call> (L=0x4123e378) at /server/ext/sol/include/sol/sol.hpp:8879
#8  sol::function_detail::upvalue_this_member_function<CLuaBaseEntity, void (CLuaBaseEntity::*)(CLuaBaseEntity*, unsigned short, sol::basic_object<sol::basic_reference<false> > const&, sol::basic_object<sol::basic_reference<false> > const&)>::call<false, false> (L=0x4123e378) at /server/ext/sol/include/sol/sol.hpp:19332
#9  0x00007f45622c8fb7 in ?? () from /lib/x86_64-linux-gnu/libluajit-5.1.so.2
#10 0x00007f45623174a4 in lua_pcall () from /lib/x86_64-linux-gnu/libluajit-5.1.so.2
#11 0x0000000000681193 in sol::basic_protected_function<sol::stack_reference, true, sol::basic_reference<false> >::luacall<true> (this=0x7ffc17f0a940, argcount=3, result_count_=-1, h=...) at /server/ext/sol/include/sol/sol.hpp:21042
#12 sol::basic_protected_function<sol::stack_reference, true, sol::basic_reference<false> >::invoke<true>(sol::types<>, std::integer_sequence<unsigned long>, long, sol::detail::protected_handler<true, sol::basic_reference<false> >&) const (this=0x7ffc17f0a940, n=3, h=...) at /server/ext/sol/include/sol/sol.hpp:21073
#13 sol::basic_protected_function<sol::stack_reference, true, sol::basic_reference<false> >::call<, CLuaBaseEntity, CLuaBaseEntity, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [10]> >&>(CLuaBaseEntity&&, CLuaBaseEntity&&, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [10]> >&) const (this=0x7ffc17f0a940, args=..., args=..., args=...) at /server/ext/sol/include/sol/sol.hpp:20981
#14 sol::table_proxy<sol::basic_table_core<true, sol::basic_reference<false> >&, std::tuple<char const (&) [3], char const (&) [8], char const (&) [12], char const (&) [19], char const (&) [10]> >::call<, CLuaBaseEntity, CLuaBaseEntity, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [10]> >&>(CLuaBaseEntity&&, CLuaBaseEntity&&, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [10]> >&) (this=0x7ffc17f0aa80, args=..., args=..., args=...) at /server/ext/sol/include/sol/sol.hpp:24839
#15 sol::table_proxy<sol::basic_table_core<true, sol::basic_reference<false> >&, std::tuple<char const (&) [3], char const (&) [8], char const (&) [12], char const (&) [19], char const (&) [10]> >::operator()<CLuaBaseEntity, CLuaBaseEntity, sol::table_proxy<sol::basic_table_core<false, sol::basic_reference<false> >, std::tuple<char const (&) [10]> >&> (this=0x7ffc17f0aa80, args=..., args=..., args=...) at /server/ext/sol/include/sol/sol.hpp:24844
#16 luautils::OnTrigger (PChar=0x7f455d1d7430, PNpc=<optimized out>) at /server/src/map/lua/luautils.cpp:1917
#17 0x00000000004792e6 in CAIContainer::Trigger (this=0x1ee66720, player=0x7f455d1d7430) at /server/src/map/ai/ai_container.cpp:139
#18 0x00000000007ca60e in SmallPacket0x01A (PSession=<optimized out>, PChar=0x7f455d1d7430, data=...) at /server/src/map/packet_system.cpp:824
#19 0x0000000000803350 in std::_Function_handler<void (map_session_data_t*, CCharEntity*, CBasicPacket), void (*)(map_session_data_t*, CCharEntity*, CBasicPacket)>::_M_invoke(std::_Any_data const&, map_session_data_t*&&, CCharEntity*&&, CBasicPacket&&) (__functor=..., __args=..., __args=..., __args=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/std_function.h:300
#20 0x00000000007aa3ea in std::function<void (map_session_data_t*, CCharEntity*, CBasicPacket)>::operator()(map_session_data_t*, CCharEntity*, CBasicPacket) const (this=<optimized out>, __args=..., __args=..., __args=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/std_function.h:688
#21 parse (buff=0x7f455d17ff00 "\203", buffsize=<optimized out>, from=<optimized out>, map_session_data=0x7f455d1858b0) at /server/src/map/map.cpp:746
#22 do_sockets (rfd=<optimized out>, next=...) at /server/src/map/map.cpp:512
#23 0x000000000041e0eb in main (argc=1, argv=<optimized out>) at /server/src/common/kernel.cpp:278
```

**Comment by WinterSolstice8** [comment]
Any dump for this?

**Comment by kaincenteno** [comment]
@WinterSolstice8  no dumps. Should I start attaching dumps from crash logs?

**Comment by Dev-Roland** [comment]
Genkai 2 is also affected/causes the same crash, and the greater genkais could potentially be affected as well.

**Comment by Xaver-DaRed** [comment]
#2090 

No LB quest was the cause of the crash. Or at the very least, I wasn't able to crash my server.

**Comment by WinterSolstice8** [comment]
Was this fixed since #2354? Symptom appears to be the same at a glance.

**Comment by WinterSolstice8** [comment]
On second glance:
>#9  0x00007f45622c8fb7 in ?? () from /lib/x86_64-linux-gnu/libluajit-5.1.so.2

Probably fixed by #2354
